src/js/base/app.js

```diff
@@ -1,9 +1,19 @@
-import { bind, isArray, noop, uniqueId } from 'underscore';
+import { bind, isArray, noop, uniqueId, extend } from 'underscore';
 import { App } from 'marionette.toolkit';
 
 import handleErrors from 'js/utils/handle-errors';
 
 export default App.extend({
+  // TODO: Move this to marionette.toolkit
+  restart(options) {
+    const state = this.getState().attributes;
+
+    this._isRestarting = true;
+    this.stop().start(extend({ state }, options));
```

Comment:
This is really the only change.. previously was just `this.stop().start({ state });`